Item07-17 - moving to modern cxx/Item08 - nullptr/main.cpp

```diff
@@ -33,8 +36,8 @@ int main ()
 
     auto x = iDefinitelyReturnAPointer (false);
     if (x == 0) {
-        //... ambiguous comparison of x. is x really a pointer???
-        // this isn't very safe...
+        // non-ambiguous comparison of x. x will be int, because of the implicit cast
```

Comment:
The main idea behind this code was not specifically what the type of the `x` variable is in this specific case, but rather the fact that the code often times can be misleading. Using the `auto` specifier for automatic type deduction can hide this detail from the API user. 
Other times, we might have a Singleton utility class, which has a dedicated referenced representative null value, and an overloaded `operator== (std::nullptr)` that checks if the object points to the custom null reference. But it could also implement an equality check for integers. In that case, `object == 0` would be very misleading - hence the lesson of always useing `std::nullptr` when checking if an object is "null".